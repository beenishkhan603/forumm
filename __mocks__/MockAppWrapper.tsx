import { AuthContext } from '@libs/useAuth'
import { MockedProvider, MockedResponse } from '@apollo/client/testing'
import { ProfileInfo } from '@type/Hooks/useAuth/Auth.type'
import ALL_MOCKS from './graphql/mocks'

export const defaultMockProfile: ProfileInfo = {
  company: '448 Studio',
  email: 'test@email.com',
  fullName: 'John Doe',
  university: '448 Studio',
  profileImageUrl: '/test.png',
  userId: '123',
  groups: [],
}

const MockAppWrapper = ({
  children,
  mocks,
  mockProfile = defaultMockProfile,
}: {
  mockProfile?: Partial<ProfileInfo>
  children: any
  mocks?: readonly MockedResponse<Record<string, any>>[] | undefined
}) => {
  const Layout = children?.type?.Layout ?? null

  const GQL_MOCKS: MockedResponse[] = ALL_MOCKS

  if (mocks) mocks.forEach((m) => GQL_MOCKS.push(m))
  return (
    <MockedProvider
      mocks={GQL_MOCKS}
      defaultOptions={{
        query: {
          errorPolicy: 'all',
          fetchPolicy: 'cache-first',
        },
        watchQuery: {
          errorPolicy: 'all',
          fetchPolicy: 'cache-and-network',
        },
        mutate: {
          errorPolicy: 'all',
        },
      }}
    >
      <AuthContext.Provider
        value={{
          profile: mockProfile
            ? { ...defaultMockProfile, ...mockProfile }
            : defaultMockProfile,
          setProfile: () => {},
        }}
      >
        <div id="modal-root" />
        {Layout ? <Layout>{children}</Layout> : children}
      </AuthContext.Provider>
    </MockedProvider>
  )
}

export default MockAppWrapper
