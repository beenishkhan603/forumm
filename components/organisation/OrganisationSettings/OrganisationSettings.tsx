import { useEffect, useState } from 'react'
import { useMutation, useQuery } from '@apollo/client'
import { UPDATE_ORGANISATION } from '@graphql/organisation/updateOrganisation'
import { GET_ORGANISATION_BY_NAME } from '@graphql/organisation/getOrganisationByName'
import Image from 'next/image'
import { Button } from '@components/inputs/Button'
import Box from '@components/base/Box'
import { FileInput } from '@components/inputs/FileInput'
import { ColorPicker } from '@components/inputs/ColorPicker'
import useFileUploader from '@libs/useFileUploader'
import TickIcon from '@public/images/TickIcon.svg'
import { useTheme } from '@libs/useTheme'
import { RadioFieldInput } from '@components/inputs/RadioFieldInput'
import { useBlackBaud } from '@libs/useBlackBaud'
import { useAuth } from '@libs/useAuth'
import { useRouter } from 'next/router'
import Modal from '@components/base/Modal'
import { BiCog, BiRightArrowAlt } from 'react-icons/bi'
import { CREATE_MERCHANT } from '@graphql/users/createMerchant'
import { GET_IS_MERCHANT } from '@graphql/users/getMerchantAccount'
import { ErrorType } from '@libs/ErrorHandler/error.type'

export default function OrganisationSettings({ name }: { name: string }) {
  const uploadFile = useFileUploader()
  const { initiateOAuth, onCallback, accessToken, completed } = useBlackBaud()

  const [updateOrganisation] = useMutation(UPDATE_ORGANISATION)
  const [logo, setLogo] = useState<string | File | undefined>()
  const [mainColour, setMainColour] = useState<string | undefined>()
  const [orgCurrency, setOrgcurrency] = useState<string>('GBP')
  const [isLoading, setIsLoading] = useState<boolean>(false)
  const [updated, setUpdated] = useState<boolean>(false)
  const { updateHighlightColour } = useTheme()
  const { isOrganizer, isAdmin, profile } = useAuth()
  const router = useRouter()

  if (!isOrganizer && !isAdmin) {
    router.push('/')
  }

  const { data: organisationData } = useQuery(GET_ORGANISATION_BY_NAME, {
    variables: {
      name,
    },
    skip: !name,
  })

  useEffect(() => {
    if (organisationData?.getOrganisationByName) {
      const {
        mainColour: currentMainColour,
        logoImage,
        currency,
      } = organisationData?.getOrganisationByName
      if (logoImage) setLogo(logoImage)
      if (currentMainColour) setMainColour(currentMainColour)
      if (currency) setOrgcurrency(currency)
    }
  }, [organisationData])

  useEffect(() => {
    // Check if the URL contains the OAuth callback parameters
    if (
      window.location.search.includes('code') &&
      window.location.search.includes('state')
    ) {
      onCallback(window.location.href, name)
    }
  }, [onCallback, name])

  const handleUpdateOrganisation = async () => {
    const variables = {
      name,
      mainColour,
      logoImage: logo,
      currency: orgCurrency,
    } as {
      name: string
      mainColour?: string
      logoImage?: string
      currency?: string
    }
    if (typeof logo === 'string') {
      variables.logoImage = logo
    }
    setIsLoading(true)
    await updateOrganisation({
      variables,
    })
    setIsLoading(false)
    if (variables.mainColour !== undefined && updateHighlightColour) {
      updateHighlightColour(variables?.mainColour)
    }

    setUpdated(true)
    setTimeout(() => {
      setUpdated(false)
    }, 3000)
  }

  const handleBlackBaud = () => {
    initiateOAuth()
  }

  const publicUrl =
    window.location.origin +
    '/org/' +
    organisationData?.getOrganisationByName?.url
  const previewLogo = typeof logo === 'string' ? logo : '/images/ForummLogo.svg'
  const connectedToBlackBaud =
    completed || organisationData?.getOrganisationByName?.blackBaudAccessToken

  const [errors, setErrors] = useState<ErrorType | undefined>(undefined)
  const [createMerchant] = useMutation(CREATE_MERCHANT)
  const { data: merchantQueryData } = useQuery(GET_IS_MERCHANT, {
    variables: {
      input: { userId: profile?.userId, company: profile?.company },
    },
  })

  const merchantData = (() => {
    if (!merchantQueryData) return {}

    const { merchantAccountExists, required, chargesEnabled } =
      merchantQueryData.getMerchantAccount

    return {
      isMerchant: merchantAccountExists,
      onboardingComplete: required && required.length < 1,
      chargesEnabled,
      isMerchantComplete:
        merchantAccountExists &&
        chargesEnabled &&
        required &&
        required.length < 1,
    }
  })()

  const { query } = useRouter()

  const [showConfirmationModal, setShowConfirmationModal] = useState(false)
  const [showStripeModal, setShowStripeModal] = useState(false)
  const [isLinkButtonLoading, setLinkButtonLoading] = useState(false)

  const handleStripeConnect = async (
    e: React.MouseEvent,
    isInitialSetup: boolean = true
  ) => {
    e.preventDefault()
    setLinkButtonLoading(true)

    if (!profile?.userId)
      throw new Error('You must be logged in to create a merchant account.')

    await createMerchant({
      variables: {
        input: {
          userId: profile.userId,
        },
      },
    })
      .then((data) => {
        if (data.errors) throw data.errors
        window.location.href = data.data?.createMerchant.redirectUrl!
        setLinkButtonLoading(false)
      })
      .catch((err) => {
        setErrors({ message: err[0].message })
        setLinkButtonLoading(false)
      })
  }

  useEffect(() => {
    if (query.stripeconfirmed === 'true') setShowConfirmationModal(true)
  }, [query])

  const successModal = (
    <Modal
      title="Stripe Connect Confirmation"
      show={showConfirmationModal}
      setShow={setShowConfirmationModal}
    >
      <Box className="flex flex-col mx-10 text-sm font-light">
        <Box className="flex flex-col items-center">
          <Box className="py-2 flex font-bold text-2xl">
            You have successfully linked your Stripe account.
          </Box>
        </Box>
      </Box>
    </Modal>
  )

  const stripeModal = (
    <Modal
      title="Stripe Settings"
      show={showStripeModal}
      setShow={setShowStripeModal}
    >
      <Box className="flex flex-col mx-10 text-sm font-light">
        <Box className="flex flex-col">
          Once all of the following requirements are met, you can begin to
          accept payments through Stripe Connect.
          <Box className="py-2 flex font-bold mt-2">
            Stripe Account Created:{' '}
            <span className="pl-2">
              {merchantData.isMerchant ? '✔' : '❌'}
            </span>
          </Box>
          <Box className="py-2 flex font-bold">
            Onboarding Complete:{' '}
            <span className="pl-2">
              {merchantData.onboardingComplete ? '✔' : '❌'}
            </span>
          </Box>
          <Box className="py-2 flex font-bold">
            Stripe Account Activated:{' '}
            <span className="pl-2">
              {merchantData.chargesEnabled ? '✔' : '❌'}
            </span>
          </Box>
          {!merchantData.isMerchantComplete && (
            <Box className="py-2 flex">
              Please complete the stripe connect onboarding process.
              <Button
                title="Setup"
                icon={<BiRightArrowAlt />}
                iconPos="end"
                className="ml-2"
                buttonType="button"
                type="tertiary"
                loading={isLinkButtonLoading}
                onClick={(e) => handleStripeConnect(e, false)}
              />
            </Box>
          )}
        </Box>
      </Box>
    </Modal>
  )

  const generateStripeConnectButton = () => {
    const connectedUI = (
      <Button
        title="Stripe"
        icon={<BiCog />}
        iconPos="end"
        className="ml-2"
        buttonType="button"
        loading={isLinkButtonLoading}
        onClick={() => setShowStripeModal(true)}
      />
    )

    if (merchantData.isMerchantComplete) return connectedUI

    if (!merchantData.isMerchant)
      return (
        <Button
          title="Connect Stripe"
          className="ml-2"
          buttonType="button"
          loading={isLinkButtonLoading}
          onClick={handleStripeConnect}
        />
      )

    if (!merchantData.chargesEnabled || !merchantData.onboardingComplete)
      return (
        <Button
          title="Stripe"
          icon={<BiCog />}
          iconPos="end"
          className="ml-2"
          buttonType="button"
          loading={isLinkButtonLoading}
          onClick={() => setShowStripeModal(true)}
        />
      )

    return connectedUI
  }

  return (
    <Box className="flex justify-center mb-10 mt-20">
      {successModal}
      {stripeModal}
      <Box className="flex flex-col items-center mt-14 text-white max-w-md w-full">
        <Box className="text-3xl">Organisation Settings</Box>
        <Box className="pt-2 justify-center flex flex-col">
          <Box className="relative justify-center  mb-8">
            <Image
              src={previewLogo}
              width={160}
              height={160}
              alt="Profile Image"
              className="w-40 h-40 bg-contain bg-center rounded-full object-cover mt-8 ml-4"
            />
            <Box className="absolute top-10 right-2">
              <FileInput
                uploadFile={uploadFile}
                value={logo}
                onChange={(data) => {
                  setLogo(data)
                }}
                minified
              />
            </Box>
          </Box>
          <ColorPicker
            className="font-medium"
            value={mainColour}
            label="Main Organisation Colour:"
            onChange={(newColour) => {
              setMainColour(newColour)
            }}
          />
          <Box className="pb-2 mt-10">
            <RadioFieldInput
              label="Select currency:"
              testid={'currency-radio'}
              options={['USD', 'CAD', 'EUR', 'GBP']}
              className="font-medium"
              itemClassName="text-sm mt-4"
              value={orgCurrency}
              onChange={(currency) => {
                setOrgcurrency(currency)
              }}
            />
          </Box>

          {generateStripeConnectButton()}
          <Box className="mt-10">
            <p className="font-medium mb-1">Organisation Public URL:</p>
            <p>
              <a
                target="_blank"
                rel="noopener noreferrer"
                className="text-[#0052DD]"
                href={publicUrl}
              >
                {publicUrl}
              </a>
            </p>
          </Box>
        </Box>
        {updated && (
          <Box
            className={`transition-all !text-white p-4 flex items-center justify-center mt-4 bg-green-400 pointer-events-none rounded text-center`}
          >
            Your organisation has been updated!
            <TickIcon className="inline-block h-8 w-8 ml-2" />
          </Box>
        )}
        {connectedToBlackBaud && (
          <span
            onClick={handleBlackBaud}
            className="w-full mt-6 text-center text-xs cursor-pointer"
          >
            <b>Connected to Blackbaud.</b> To keep your session secure, it may
            expire after some time. Please, use the button below to refresh your
            session when needed.
          </span>
        )}
        <Button
          type="secondary"
          loading={isLoading}
          onClick={handleBlackBaud}
          className="w-full mt-2"
          title={!connectedToBlackBaud ? 'Connect to BlackBaud' : 'Reconnect'}
        />
        <Button
          loading={isLoading}
          onClick={handleUpdateOrganisation}
          className="w-full mt-8"
          title="Save Changes"
        />
      </Box>
    </Box>
  )
}
