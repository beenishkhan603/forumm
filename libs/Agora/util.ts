import { IAgoraToForummUser } from '@type/Broadcasting/Agora.type'
import { ProfileInfo } from '@type/Hooks/useAuth/Auth.type'
import { IAgoraRTCRemoteUser } from 'agora-rtc-sdk-ng'

export interface IAgoraRemoteSpeaker extends IAgoraRTCRemoteUser {
  volume?: number
}

export enum AgoraListeners {
  // 200
  'REMOTE_USERS_CHANGE' = 'remoteUsersChange',
  'LOCAL_VIDEO_TRACK' = 'localVideoTrack',
  'LOCAL_AUDIO_TRACK' = 'localAudioTrack',
  'LOCAL_SCREEN_TRACK' = 'localScreenTrack',
  'SCREENSHARE_TOGGLED' = 'screenShareToggled',
  'MICROPHONE_TOGGLED' = 'microphoneToggled',
  'CAMERA_TOGGLED' = 'cameraToggled',
  'TOKEN_RENEWED' = 'tokenRenewed',

  // 400
  // Screenshare Cancelled
  // @return bool indicating if the screenshare request has been cancelled.
  'SCREENSHARE_CANCELLED' = 'screenShareCancelled',
  'NETWORK_ISSUES' = 'networkIssues',
  'NETWORK_DISCONNECTED' = 'networkDisconnected',
  'LOGGER_REQUESTED' = 'loggerRequested',
}

export const SpeakerJoinedGreenRoomEvent = (data: {
  channel: string
  token?: string
  userId?: string
}) => {
  return new CustomEvent('speaker_joined_green_room', { detail: data })
}

export const AgoraInterfaceError = class extends Error {
  constructor(message: string) {
    super(message)
    this.message = message
    this.name = '[AgoraInterfaceError]'
  }
}

export const mapAgoraUserToForummUser = (
  user: IAgoraRTCRemoteUser,
  forummUsers: ProfileInfo[],
  localProfile?: ProfileInfo
): IAgoraToForummUser | undefined => {
  if (!user || !user.uid) return
  const mappedUser =
    forummUsers?.find((u) => {
      return u.userId === user.uid.toString().slice(0, 36)
    }) ?? {}

  const payload = {
    ...user,
    ...mappedUser,

    isLocalTrack: user.uid.toString() === localProfile?.userId?.toString(),
    isScreen:
      user.uid.toString().includes('>screen') ||
      user.uid.toString().includes('>local_screen'),
    isLocalScreen:
      user.uid.toString() === localProfile?.userId?.toString() ||
      user.uid.toString().includes('>local_screen'),
  }
  return payload
}

export const playOnElement = (stream: MediaStream, element: HTMLDivElement) => {
  const videoEl = document.createElement('video')
  videoEl.setAttribute('playsinline', '')
  videoEl.setAttribute('autoplay', '')
  videoEl.setAttribute('muted', '')

  if (!!stream) {
    videoEl.srcObject =
      (stream as unknown as { stream: MediaStream })?.stream ?? undefined

    element.appendChild(videoEl)
  }
  return () => {
    try {
      element.removeChild(videoEl)
    } catch (e) {
      console.error('PlayOnElement', e)
    }
  }
}
