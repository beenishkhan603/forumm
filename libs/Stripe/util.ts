import { PaymentMethod } from '@stripe/stripe-js'
import { EUROPE_ECONOMIC_AREA, STRIPE_PAYMENT_FEES } from './processor-fees'

export const getBankingCountryByPaymentMethod = (
  paymentMethod: PaymentMethod
) => {
  if (!paymentMethod) return undefined
  const { type, card } = paymentMethod
  return card?.country ?? undefined
}

export const getProcessorFees = (regionCode?: string) => {
  if (!regionCode) return STRIPE_PAYMENT_FEES.INTERNATIONAL
  if (regionCode === 'GB') return STRIPE_PAYMENT_FEES.GBP
  if (EUROPE_ECONOMIC_AREA.includes(regionCode)) return STRIPE_PAYMENT_FEES.EEA
  return STRIPE_PAYMENT_FEES.INTERNATIONAL
}

export const getDonationTotal = ({
  amount,
  platformFee,
  regionCode,
  coverPlatformFee,
  coverProcessorFee,
}: {
  amount: number
  platformFee: number
  regionCode?: string
  coverPlatformFee?: boolean
  coverProcessorFee?: boolean
}) => {
  const { flat, percent } = getProcessorFees(regionCode)
  const processorFee = percent / 100
  const processorFixedFeeCents = flat

  let donationAmountCents = amount

  let totalDonationCents = 0

  let processorFeeAmountCents = 0

  const platformFeeAmountCents = donationAmountCents * (platformFee / 100)

  if (coverProcessorFee && coverPlatformFee) {
    totalDonationCents = donationAmountCents + platformFeeAmountCents

    processorFeeAmountCents =
      (totalDonationCents + processorFixedFeeCents * 2) * processorFee +
      processorFixedFeeCents

    totalDonationCents += Math.ceil(processorFeeAmountCents)
  }

  if (!coverProcessorFee && !coverPlatformFee) {
    totalDonationCents = donationAmountCents

    processorFeeAmountCents =
      totalDonationCents * processorFee + processorFixedFeeCents

    donationAmountCents =
      totalDonationCents -
      platformFeeAmountCents -
      Math.ceil(processorFeeAmountCents)
  }

  return {
    donation: donationAmountCents, // Amount going to charity (in cents)
    platformFee: platformFeeAmountCents, // Platform fee (in cents)
    processorFee: Math.ceil(processorFeeAmountCents), // Payment processor fee (in cents)
    total: totalDonationCents, // Total donation amount (in cents)
    regionCode,
  }
}
