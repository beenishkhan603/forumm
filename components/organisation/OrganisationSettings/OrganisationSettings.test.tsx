import { useJestRouter } from '@libs/Utility/Test/util'
import { render, waitFor, screen } from '@testing-library/react'
import MockAppWrapper, { defaultMockProfile } from '__mocks__/MockAppWrapper'
import OrganisationSettings from './OrganisationSettings'

const pushSpy = jest.fn()

jest.mock('next/router', () => {
  const realModule = jest.requireActual('next/router')
  return {
    ...realModule,
    useRouter() {
      return {
        query: '',
        pathname: '',
        push: pushSpy,
      }
    },
  }
})
describe('Organisation Settings', () => {
  beforeEach(() => {
    pushSpy.mockClear()
  })

  test('Snapshot', () => {
    const { container } = render(
      <MockAppWrapper>
        <OrganisationSettings name={defaultMockProfile.company!} />
      </MockAppWrapper>
    )
    expect(container).toMatchSnapshot()
  })

  describe('User Auth', () => {
    test("SHOULD push user to dashboard WHEN they're not authorised", () => {
      render(
        <MockAppWrapper>
          <OrganisationSettings name={defaultMockProfile.company!} />
        </MockAppWrapper>
      )
      expect(pushSpy).toBeCalledWith('/')
    })

    test("SHOULD NOT push user to dashboard WHEN they're authorised (Organiser)", () => {
      render(
        <MockAppWrapper mockProfile={{ groups: ['organizer'] }}>
          <OrganisationSettings name={defaultMockProfile.company!} />
        </MockAppWrapper>
      )
      expect(pushSpy).not.toBeCalledWith('/')
    })

    test("SHOULD NOT push user to dashboard WHEN they're authorised (Admin)", () => {
      render(
        <MockAppWrapper mockProfile={{ groups: ['forumm-admin'] }}>
          <OrganisationSettings name={defaultMockProfile.company!} />
        </MockAppWrapper>
      )
      expect(pushSpy).not.toBeCalledWith('/')
    })
  })
})
