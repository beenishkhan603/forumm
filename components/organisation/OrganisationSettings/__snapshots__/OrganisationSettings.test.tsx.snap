// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organisation Settings Snapshot 1`] = `
<div>
  <div
    id="modal-root"
  />
  <div
    class="flex justify-center mb-10 mt-20"
    style="color: rgb(24, 26, 32);"
  >
    <div
      class="flex flex-col items-center mt-14 text-white max-w-md w-full"
      style="color: rgb(24, 26, 32);"
    >
      <div
        class="text-3xl"
        style="color: rgb(24, 26, 32);"
      >
        Organisation Settings
      </div>
      <div
        class="pt-2 justify-center flex flex-col"
        style="color: rgb(24, 26, 32);"
      >
        <div
          class="relative justify-center mb-8"
          style="color: rgb(24, 26, 32);"
        >
          <img
            alt="Profile Image"
            class="w-40 h-40 bg-contain bg-center rounded-full object-cover mt-8 ml-4"
            data-nimg="1"
            decoding="async"
            height="160"
            loading="lazy"
            src="/images/ForummLogo.svg"
            srcset="/images/ForummLogo.svg 1x, /images/ForummLogo.svg 2x"
            style="color: transparent;"
            width="160"
          />
          <div
            class="absolute top-10 right-2"
            style="color: rgb(24, 26, 32);"
          >
            <label
              class="flex flex-col space-y-1 "
              for="formFile"
            >
              <div
                class=" text-white rounded-full p-2 flex items-center cursor-pointer hover:animate-heartbeat"
                style="background-color: rgb(55, 99, 233);"
              >
                <svg
                  class="h-4 w-4 inline-block"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  style="color: white;"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04a.996.996 0 000-1.41l-2.34-2.34a.996.996 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                  />
                </svg>
              </div>
              <input
                accept="image/*,.mp4"
                class="hidden"
                type="file"
              />
            </label>
          </div>
        </div>
        <label
          class="flex flex-col space-y-1 font-medium"
        >
          <span
            class="transition-all "
          >
            Main Organisation Colour:
             
            <span
              class="ml-2 text-xs"
            />
          </span>
          <div
            class="relative"
            style="color: rgb(24, 26, 32);"
          >
            <svg
              class="w-5 h-5 cursor-pointer absolute my-4 mr-4 right-0"
            />
            <input
              autocomplete="on"
              class="bg-transparent select-none cursor-pointer  rounded px-4 py-4 text-sm outline-none w-full"
              placeholder="Select Color"
              style="color: rgb(255, 255, 255);"
              type="text"
              value=""
            />
          </div>
          <div
            class="transition-all opacity-0 text-red-500 text-sm font-bold"
            style="color: rgb(24, 26, 32);"
          />
        </label>
        <div
          class="pb-2 mt-10"
          style="color: rgb(24, 26, 32);"
        >
          <span
            class="flex flex-col space-y-1 font-medium"
          >
            <span
              class="-mb-3 transition-all "
            >
              <span
                class=""
              >
                Select currency:
              </span>
              <span
                class="ml-2 text-xs"
              />
            </span>
            <div
              class="flex flex-wrap"
              style="color: rgb(24, 26, 32);"
            >
              <span
                class="text-sm mt-4 mt-4 flex items-center cursor-pointer hover:opacity-90 mr-4"
              >
                <span
                  class="mr-2 text-sm text-gray-600"
                >
                  USD
                </span>
                <input
                  class="h-4 w-4 text-forumm-blue border-gray-300"
                  data-testid="currency-radio-0"
                  name="select_currency:"
                  type="radio"
                  value="USD"
                />
              </span>
              <span
                class="text-sm mt-4 mt-4 flex items-center cursor-pointer hover:opacity-90 mr-4"
              >
                <span
                  class="mr-2 text-sm text-gray-600"
                >
                  CAD
                </span>
                <input
                  class="h-4 w-4 text-forumm-blue border-gray-300"
                  data-testid="currency-radio-1"
                  name="select_currency:"
                  type="radio"
                  value="CAD"
                />
              </span>
              <span
                class="text-sm mt-4 mt-4 flex items-center cursor-pointer hover:opacity-90 mr-4"
              >
                <span
                  class="mr-2 text-sm text-gray-600"
                >
                  EUR
                </span>
                <input
                  class="h-4 w-4 text-forumm-blue border-gray-300"
                  data-testid="currency-radio-2"
                  name="select_currency:"
                  type="radio"
                  value="EUR"
                />
              </span>
              <span
                class="text-sm mt-4 mt-4 flex items-center cursor-pointer hover:opacity-90 mr-4"
              >
                <span
                  class="mr-2 text-sm text-gray-600"
                >
                  GBP
                </span>
                <input
                  checked=""
                  class="h-4 w-4 text-forumm-blue border-gray-300"
                  data-testid="currency-radio-3"
                  name="select_currency:"
                  type="radio"
                  value="GBP"
                />
              </span>
            </div>
            <div
              class="transition-all opacity-0 text-red-500 text-sm font-bold"
            />
          </span>
        </div>
        <button
          class="
      bg-forumm-blue-2 text-white relative rounded-full  px-6 py-4 text-base h-8 flex items-center hover:animate-heartbeat justify-center ml-2"
          style="background-color: rgb(55, 99, 233);"
          type="button"
        >
          <span
            class="relative z-10"
            style="color: white;"
          >
            Connect Stripe
          </span>
        </button>
        <div
          class="mt-10"
          style="color: rgb(24, 26, 32);"
        >
          <p
            class="font-medium mb-1"
          >
            Organisation Public URL:
          </p>
          <p>
            <a
              class="text-[#0052DD]"
              href="http://localhost/org/undefined"
              rel="noopener noreferrer"
              target="_blank"
            >
              http://localhost/org/undefined
            </a>
          </p>
        </div>
      </div>
      <button
        class="
      border border-forumm-menu-border text-forumm-blue-2 bg-forumm-white relative  rounded-full  px-6 py-4 text-base h-8 flex items-center hover:animate-heartbeat justify-center w-full mt-2"
        type="button"
      >
        <span
          class="relative z-10"
          style="color: white;"
        >
          Connect to BlackBaud
        </span>
      </button>
      <button
        class="
      bg-forumm-blue-2 text-white relative rounded-full  px-6 py-4 text-base h-8 flex items-center hover:animate-heartbeat justify-center w-full mt-8"
        style="background-color: rgb(55, 99, 233);"
        type="button"
      >
        <span
          class="relative z-10"
          style="color: white;"
        >
          Save Changes
        </span>
      </button>
    </div>
  </div>
</div>
`;
